trio/tests/test_file_io.py

```diff
@@ -0,0 +1,169 @@
+import io
+import tempfile
+
+import pytest
+from unittest import mock
+from unittest.mock import patch, sentinel
```

Comment:
I dunno. If that's unit testing, then I don't like unit testing. I want to write APIs where the relationship between the inputs and outputs is well-defined, and I want tests that check that contract is fulfilled. If other units get used internally then that's an internal implementation detail, and writing tests to make sure that internal implementation details haven't changed sounds like the opposite of helpful; I want my tests to make it *easier* to change internal implementation details!